cmd/libsnap-confine-private/string-utils.c

```diff
@@ -56,6 +56,22 @@ bool sc_endswith(const char *str, const char *suffix)
 	return strncmp(str - xlen + slen, suffix, xlen) == 0;
 }
 
+char *sc_strdup(const char *str)
+{
+	size_t len;
+	char *copy;
+	if (str == NULL) {
+		die("cannot duplicate NULL string");
+	}
+	len = strlen(str);
```

Comment:
`stdup()` does exactly that: strlen(str), malloc(len+1), memcpy(), why not use it and do:
```
char * sc_strdup(const char *str)
{
    char *copy = stdup(str);
    if (copy == NULL) {
        die("cannot allocate string copy (len: %zd)", len);
    }
    return copy
}
```

Reply:
Because it doesn't tell use the length :D

Reply:
Is `len` useful though? Just wondering if it helps in any way to show it in the error message.

Reply:
In case we ever want to malloc -1 bytes, I suspect it would be useful.

Reply:
In that case it makes sense indeed.